### ISSUE TITLE
Static libraries won't link (Windows 10 x64)

### ISSUE THREAD

**Issue by benkuper** [switch-joytime/joytime-input-hidapi#3]
I built the joytime-demo, then i took the 3 _static.lib files (hidapi, joytime-input-hidapi and joytime-core) and copied them with the .hpp files, linked everything in VS 2017 x64 and got unresolved errors :

```
1>JoyConModule.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: __cdecl Joytime::Controller::~Controller(void)" (__imp_??1Controller@Joytime@@QEAA@XZ) referenced in function "public: void * __cdecl Joytime::Controller::`scalar deleting destructor'(unsigned int)" (??_GController@Joytime@@QEAAPEAXI@Z)
1>JoyConModule.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) class std::vector<class Joytime::Controller,class std::allocator<class Joytime::Controller> > __cdecl Joytime::scanForControllers(void)" (__imp_?scanForControllers@Joytime@@YA?AV?$vector@VController@Joytime@@V?$allocator@VController@Joytime@@@std@@@std@@XZ) referenced in function "public: __cdecl JoyConModule::JoyConModule(class juce::String const &)" (??0JoyConModule@@QEAA@AEBVString@juce@@@Z)
```
Here are some screenshots of my configuration. I have other libraries that links well.

![image](https://user-images.githubusercontent.com/1291595/35289576-04b7b92e-0068-11e8-8cc9-b7aee1abe53e.png)

![image](https://user-images.githubusercontent.com/1291595/35289616-1ffddfa6-0068-11e8-808c-6a0c2d3d1b58.png)

The include folder in the picture above contains all the joytime-* .hpp files including the _EXPORTS.h as well as hidapi.h and EventEmitter.hpp

**Comment by facekapow** [comment]
First of all, thank you! Thanks for trying it out on something. But that certainly is strange. Are you using CMake to link it? If not, could you try that and tell me whether you're still getting the same error? (same goes for your first issue, #2)

**Comment by benkuper** [comment]
Yes, i followed your guide so everything is generated with cmake (option -G "Visual Studio 2017 Win64") and then compile with VS.
For now i don't really care about static linking but it would be cool :)
I'm trying to integrate joytime into my software Chataigne : http://benjamin.kuperberg.fr/chataigne so i can link it to almost any protocols and artistic projects.
